### PR TITLE
feat: support openapi yaml include validation

### PR DESCRIPTION
**Pull request by NickHeap2**
#### Which issue(s) does this change fix?
#3041
#3676

#### Why is this change necessary?
If you are using OpenAPI yaml specifications it makes sense to use them for the API DefinitionBody like this:
```
      DefinitionBody:
        'Fn::Transform':
          Name: AWS::Include
          Parameters:
            Location: openapi.yaml
```
Whilst this is valid in a template it is currently not supported by sam validate.

#### How does it address the issue?

This change detects this pattern and replaces the DefinitionBody with the actual contents of the file allowing the validation to succeed.
This PR also includes tests to make sure that it copes with Locations that are not strings as per #3676.

#### What side effects does this change have?

Potentially someone could be relying on the existing behaviour where it fails for this pattern to prevent that pattern being used.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [X] Write unit tests
- [X] Write/update functional tests
- [X] Write/update integration tests
- [X] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by aleguern** [comment]
I may be wrong but does it not miss a test with CORS enabled but without the DefinitionBody ?

**Comment by aleguern** [comment]
Any news on this @NickHeap2 ?

**Comment by NickHeap2** [comment]
@aleguern last time after a week the bot applied the maintainer/need-response tag and then it got reviewed.

**Review by torresxb1 — COMMENTED** [review]
Thank you very much for your contribution! 
Left a couple comments. Also, this is more of a nit, but could we make the names of tests/files specific (e.g. `...definition_body_include_local_path`, `...definition_body_include_non_string`, etc.)? Oh and, replacing `none_string` with `non_string` 🙏 🙏

File: tests/integration/testdata/validate/include_none_string/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+openapi: 3.0.0
+paths:
+  /test:
+    get:
+      summary: Get test
```
[the remark this answers is not in the document]
Comment (NickHeap2) [reply tests/integration/testdata/validate/include_none_string/openapi.yaml:6]: I left it in to make it clear that the test doesn't do the replace because it's not a simple string rather than because openapi.yaml doesn't exist. Will take it out though.

[the remark this answers is not in the document]
Comment (hawflau) [reply tests/integration/testdata/validate/include_none_string/openapi.yaml:6]: +1 for removing the file.

File: tests/unit/commands/local/lib/swagger/test_parser.py [hunk 2]
```diff
@@ -141,6 +141,39 @@ def test_payload_format_version(self):
 
         self.assertEqual(expected, result)
 
+    def test_payload_format_version_for_none(self):
```
Comment (hawflau) [inline tests/unit/commands/local/lib/swagger/test_parser.py:144]: Why is this test relevant to the change?

**Review by hawflau — CHANGES_REQUESTED** [review]

File: samcli/commands/validate/lib/sam_template_validator.py [hunk 3]
```diff
@@ -181,3 +212,11 @@ def _update_to_s3_uri(property_key, resource_property_dict, s3_uri_value="s3://b
             return
 
         resource_property_dict[property_key] = s3_uri_value
+
+    @staticmethod
+    def _replace_with_file_contents(resource_property_dict, location_prop):
```
[the remark this answers is not in the document]
Comment (qingchm) [reply samcli/commands/validate/lib/sam_template_validator.py:217]: Please add in docstring to this method as well as input parameter type hints!

**Comment by hawflau** [comment]
Hi @NickHeap2 would you be able to address the change requests?

**Comment by jfuss** [comment]
Closing as inactive.

Happy to review an updated PR from the community, if someone wants to revive the PR
